lib/icingadb/icingadb-objects.cpp

```diff
@@ -290,6 +343,146 @@ void IcingaDB::UpdateAllConfigObjects()
 			}
 		}
 
+		std::map<String, std::map<String, String>> ourContent;
+
+		for (auto& source : ourContentRaw) {
+			upqObjectType.Enqueue([&]() {
+				auto& dest (ourContent[source.first]);
+
+				for (auto& hMSet : source.second) {
+					String* key = nullptr;
+
+					for (auto& kv : hMSet) {
+						if (key) {
+							dest.emplace(std::move(*key), std::move(kv));
+							key = nullptr;
+						} else {
+							key = &kv;
+						}
+					}
+
+					hMSet.clear();
+				}
+
+				source.second.clear();
+			});
+		}
+
+		upqObjectType.Join();
+		ourContentRaw.clear();
+
+		auto& ourCheckSums (ourContent[configCheckSum]);
+		auto& ourObjects (ourContent[configObject]);
+		std::vector<String> setChecksum, setObject, delChecksum, delObject;
+
+		auto redisCurrent (redisCheckSums.begin());
+		auto redisEnd (redisCheckSums.end());
+		auto ourCurrent (ourCheckSums.begin());
+		auto ourEnd (ourCheckSums.end());
+
+		auto flushSets ([&]() {
+			setChecksum.insert(setChecksum.begin(), configCheckSum);
+			setChecksum.insert(setChecksum.begin(), "HMSET");
+			setObject.insert(setObject.begin(), configObject);
+			setObject.insert(setObject.begin(), "HMSET");
+
+			std::vector<std::vector<String>> transaction;
+
+			transaction.emplace_back(std::vector<String>{"MULTI"});
+			transaction.emplace_back(std::move(setChecksum));
+			transaction.emplace_back(std::move(setObject));
+			transaction.emplace_back(std::vector<String>{"EXEC"});
```

Comment:
Can't this be an initializer list (same in the next lambda)?

Reply:
Are you sure the HMSETs won’t be copied once more?